org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentSymbolHandler.java

```diff
@@ -171,11 +173,13 @@ private String getName(IJavaElement element) {
 	}
 
 	private Range getRange(IJavaElement element) throws JavaModelException {
-		return JDTUtils.toLocation(element, FULL_RANGE).getRange();
+		Location location = JDTUtils.toLocation(element, FULL_RANGE);
+		return location == null ? DEFAULT_RANGE : location.getRange();
 	}
 
 	private Range getSelectionRange(IJavaElement element) throws JavaModelException {
-		return JDTUtils.toLocation(element).getRange();
+		Location location = JDTUtils.toLocation(element, FULL_RANGE);
+		return location == null ? DEFAULT_RANGE : location.getRange();
 	}

```

Comment:
getRange and getSelectionRange are now strictly identical O_o

Reply:
Can we remove the call of "toLocation" twice to share the location? I don't know why the original purpose of SelectionRange and getRange

Reply:
updated

Reply:
-1 on removing that. It was implemented according to the [spec](https://github.com/Microsoft/language-server-protocol/blob/gh-pages/specification.md):

> 	/**
	 * The range enclosing this symbol not including leading/trailing whitespace but everything else
	 * like comments. This information is typically used to determine if the clients cursor is
	 * inside the symbol to reveal in the symbol in the UI.
	 */
	range: Range;

>	/**
	 * The range that should be selected and revealed when this symbol is being picked, e.g the name of a function.
	 * Must be contained by the `range`.
	 */
	selectionRange: Range;

You might break other clients by changing those ranges